src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -160,6 +157,17 @@ export const getModuleDetails = async (
     })
   );
 
+  // If we add a ProposalExecuted event to the OptimisticGovernor, we could check by both the hash and the timestamp.
+  // In its current form, this code will create issues if there are duplicate proposals.
+  let proposalExecuted = false;
+  const executionEvents = await moduleContract.queryFilter(
+    'TransactionExecuted'
+  );
+  const thisProposalExecutionEvents = executionEvents.filter(
+    event => event.args?.proposalHash === transactionsHash
```

Comment:
What happens if transactionsHash is undefined? Is that an expected case?

Reply:
It's not an expected case and would mean that no transactions were defined or that something else went wrong in building the transactionsHash. The function will not set proposalExecuted to true because it didn't find any executed proposals matching the hash of transactions found on Snapshot, which is fine and what we want.